File: Computer Programming/C++/Hashing/Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) 
    {
        unordered_map<int,int> hash;
        unordered_map<int,int> freq;
        for(auto it : arr)
        {
            hash[it] ++;
        }   

        for(auto it : hash)
        {
            freq[it.second] ++;
            if(freq[it.second] > 1)
            {
                return false;
            }
        }
        return true;
    }
};

File: Computer Programming/Python/Hashing/Unique Number of Occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """

        n = len(arr)
        mp = {}

        for i in range(n):
            if arr[i] in mp:
                mp[arr[i]] += 1
            else:
                mp[arr[i]] = 1


        freq = set()
        for count in mp.values():
            if count in freq:
                return False
            else:
                freq.add(count)
        
        return True
        
